src/src_sharpSAT/MainSolver/MainSolver.h

```diff
@@ -442,14 +442,54 @@ class CMainSolver: public CInstanceGraph
 	}
     
 	void translateLiterals(const vector<int> varTranslation) {
-		for (int i = 0; i < litNodes.size(); i++) {
-			if (litNodes[i]->getVal() < 0)
-			{
-				litNodes[i]->setVal(-1 * varTranslation[-1 * litNodes[i]->getVal()]);
-			}
-			else
+		set<int> nodesSeen;
+		queue<DTNode *> openList;
+		openList.push(decStack.top().getDTNode());
+
+		int rootID = openList.front()->getID();
+
+		DTNode *node;
+		while (!openList.empty())
+		{
+			node = openList.front();
+			openList.pop();
+
+			int node_id = node->getID();
+
+			if (nodesSeen.find(node_id) == nodesSeen.end())
 			{
-				litNodes[i]->setVal(varTranslation[litNodes[i]->getVal()]);
+				// Make sure we don't add this twice
+				nodesSeen.insert(node_id);
+
+				// Add the children to the open list
+				set<DTNode *>::iterator it;
+				for (it = node->getChildrenBegin(); it
+						!= node->getChildrenEnd(); it++)
+				{
+					openList.push(*it);
+				}
+
+				// process the node
+				if (DT_LIT == node->getType())
+				{
+					if (node->getVal() < 0)
+					{
+						node->setVal(-1 * varTranslation[-1 * node->getVal()]);
+					}
+					else
+					{
+						node->setVal(varTranslation[node->getVal()]);
+					}
+				}
+
+				if (DT_OR == node->getType())
+				{
+					if (node->choiceVar) {
+						node->choiceVar = varTranslation[node->choiceVar];
```

Comment:
It's this line I'm concerned with. `varTranslation` comes in as `vector<int>`, and while the other instances are `int`, this one comes in as `unsigned`.

Reply:
Oh indeed ! I had missed this one !
I went all the way and fixed the translation vector to contain unsigned values. i'm fairly confident that it's correct because I can compile sucessfully with `-Wsign-promo -Wsign-conversion -Wsign-compare -Werror `